### PR TITLE
Feature/verification

### PR DESCRIPTION
**Pull request by AKutusenko**
## Describe your changes
- I have done verification server that can add verification data to database. The document from verification data will be uploaded to aws s3 service and the database will store the file link and the file name in User table.

Adding the database schema below

image

## Issue ticket code (and/or) and link
- [Link to JIRA ticket](https://testhomework4.atlassian.net/jira/software/projects/AG/boards/6?assignee=5fd26b3708757f006e4d61d4&selectedIssue=AG-50)

### **General**
- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Backend
- [x] Swagger documentation updated
- [x] Database requests are optimized and not redundant
- [ ] Unit tests written
- [x] use ConfigService instead of process.env
- [ ] use transactions if there is a call chain that mutates data in different tables
- [ ] use @index decorator for frequently requested data

### CODE REVIEW

**Comment by lyudad** [comment]
I dont agree with your schema, why we need verification table if its connection with user table 1-to-1, same with verification file, we have only file, if its 1-to-1, why we need a separate table. with each select you need 2-3 join

**Comment by AKutusenko** [comment]
> I dont agree with your schema, why we need verification table if its connection with user table 1-to-1, same with verification file, we have only file, if its 1-to-1, why we need a separate table. with each select you need 2-3 join

Yes I tried to keep a balance between data normalization and query performance. I assumed that the amount of verification data can be expanded and the number of files can be increased from one, so I implemented this structure. If we don't need scaling, we can put everything in one table.

**Comment by lyudad** [comment]
I prefere 1 table, I dont think it make sense make it complicated

File: src/modules/user/user.controller.ts [hunk 1]
```diff
@@ -18,4 +18,30 @@ export class UserController {
   create(@Body() createUserDto: CreateUserDto): Promise<{ user: User; token: string }> {
     return this.userService.create(createUserDto);
   }
+
+  @ApiOperation({ summary: 'Getting all users' })
+  @ApiResponse({ status: 200, description: 'OK' })
+  @ApiResponse({ status: 404, description: 'Not found' })
+  @Get('/')
+  async getAllUsers(): Promise<User[] | []> {
```
Comment (lyudad) [inline src/modules/user/user.controller.ts]: не вижу зщиты по токену, или пропустила я или добавить тебе

File: src/modules/user/user.controller.ts [hunk 2]
```diff
@@ -18,4 +18,30 @@ export class UserController {
...
+    } catch (error) {
+      throw error;
+    }
+  }
+
+  @ApiOperation({ summary: 'Getting user by id' })
+  @ApiResponse({ status: 200, description: 'OK' })
+  @ApiResponse({ status: 404, description: 'Not found' })
+  @Post('/id')
+  async getUserByUd(@Body() body: { id: string }): Promise<User> {
```
Comment (lyudad) [inline src/modules/user/user.controller.ts:43]: тоже самое, это закрытые данные, не контроллер не ендпоинт на сколько я вижу не закрыт

File: src/modules/verification/verification.service.ts [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+      user.identity = identity;
+      user.status = status;
+      user.street = street;
+      user.city = city;
+      user.state = state;
+      user.zip = zip;
+      user.country = country;
+      user.phone = phone;
+      user.fileName = fileData.fileName;
+      user.fileUrl = fileData.fileUrl;
```
Comment (lyudad) [inline src/modules/verification/verification.service.ts]: если супер горит закрыть таску, то это можно скиптьнуть, но я бы искала каккой то вариант по красивее. ну типа будет 50 полей ты их все будешь присваивать, так себе. банальный пример который я вижу это через цикл. можно еще что то в нете погуглить, может по лучше цикла

File: src/modules/user/user.controller.ts (hunk 1 above)
Comment (AKutusenko) [reply src/modules/user/user.controller.ts]: fixed

File: src/modules/user/user.controller.ts (hunk 2 above)
Comment (AKutusenko) [reply src/modules/user/user.controller.ts:43]: fixed

File: src/modules/verification/verification.service.ts (hunk 3 above)
Comment (AKutusenko) [reply src/modules/verification/verification.service.ts]: fixed

File: src/modules/user/user.controller.ts (hunk 1 above)
Comment (lyudad) [reply src/modules/user/user.controller.ts]: а почему ты весь контроллер не поставил под гард а только ендпоинты? я не нашла ендпоинтов в этом контроллере которые должны быть открыты, или я ошибаюсь?

File: src/modules/user/user.service.ts [hunk 4]
```diff
@@ -52,6 +52,36 @@ export class UserService {
...
+      if (!user) {
+        throw new NotFoundException('Not found');
+      }
+      return user;
+    } catch (error) {
+      throw error;
+    }
+  }
+
+  async getAll(): Promise<User[] | []> {
```
Comment (lyudad) [inline src/modules/user/user.service.ts:69]: а разве просто User[] не ок? просто даже если вернется пустой масив, тс должен понять что это тип юзера

Comment (AKutusenko) [reply src/modules/user/user.service.ts:69]: fixed

File: src/modules/user/user.controller.ts (hunk 1 above)
Comment (AKutusenko) [reply src/modules/user/user.controller.ts]: fixed

**Review by lyudad — APPROVED** [review]

**Pull request merged**